SimulationAgent/DeviceConnection/Connect.cs

```diff
@@ -49,6 +49,8 @@ public void Setup(IDeviceConnectionActor context, string deviceId, DeviceModel d
                 var now = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
                 await this.context.Client.ConnectAsync();
                 await this.context.Client.RegisterMethodsForDeviceAsync(this.deviceModel.CloudToDeviceMethods, this.context.DeviceState, this.context.DeviceProperties);
+                await this.context.Client.RegisterDesiredPropertiesUpdateAsync(this.context.DeviceProperties);
```

Comment:
nit: should we consider 'Updates' (plural) instead of 'Update' in the name ? since this is per device should it be ResisterDesiredPropertiesUpdatesForDeviceAsync or something similar ?

Reply:
Interesting to consider. I was actually wondering if we should keep "property" singular based on the DeviceClient API: `SetDesiredPropertyUpdateCallbackAsync`

Since this isn't the callback though, it does make sense to me to make it plural for "updates"

Reply:
My 2 cents.
I like what you have since these are going to be called per device any ways, so it is update per device. But i'm ok with whatever you decide.